Add README to Html.Abstractions and update Components README (#27833)
dotnet/aspnetcore: src/Components/README.md

### Build

> To build this specific project from source, you can follow the instructions [on building a subset of the code](../../BuildFromSource.md#building-a-subset-of-the-code).

Reviewer: ```suggestion
To build this specific project from source, follow the instructions [on building a subset of the code](../../BuildFromSource.md#building-a-subset-of-the-code).
```

> Note, you will also need to run the `build` command above in the command line before building in VS to ensure that the Web.JS dependency is built.

Reviewer: ```suggestion
**Note:** You also need to run the preceding `build` command in the command line before building in VS to ensure that the Web.JS dependency is built.
```

Each app server mounts the same `BasicTestApp` application under each scenario.

> To run the tests for this project, you can [run the tests on the command line](../../BuildFromSource.md#running-tests-on-command-line) in this directory.

Reviewer: ```suggestion
To run the tests for this project, [run the tests on the command line](../../BuildFromSource.md#running-tests-on-command-line) in this directory.
```
